db/src/main/java/org/odk/collect/db/sqlite/SqlQuery.kt

```diff
@@ -9,8 +9,20 @@ data class SqlQuery(
 
 fun Query.toSql(): SqlQuery {
     return when (this) {
-        is Query.Eq -> SqlQuery("$column = ?", arrayOf(value))
-        is Query.NotEq -> SqlQuery("$column != ?", arrayOf(value))
+        is Query.Eq -> {
+            if (value.toDoubleOrNull() != null) {
```

Comment:
The general approach here is that if we have a "numeric" (a String that looks like an Int or a Double) value for the query, we convert both to double so we end up with essentially `Double = Double` right? 

One thing that feels off to me is that we'll explicitly have the type (an Int or a Double) in `LocalEntitiyFilterStrategy` and then we're "erasing" it by converting to a String and then reconverting based on a heuristic. Instead could we add `DoubleEq` and `DoubleNotEq` types to `Query` (I guess the existing ones would be `StringEq`)? Then we'd just lift `Int` to Double when creating a `Query` and `toSql` wouldn't have to make any guesses.

Reply:
The test case to write to drive out the above I'm realizing is a filter expression like `id = '1234'`. This implementation would match an `id` value of `'1234.0'` (as both get converted to doubles) whereas only the exact string `'1234'` should match.

Reply:
Done.